array_api_tests/test_creation_functions.py

```diff
@@ -355,13 +355,14 @@ def test_eye(n_rows, n_cols, kw):
     _n_cols = n_rows if n_cols is None else n_cols
     ph.assert_shape("eye", out_shape=out.shape, expected=(n_rows, _n_cols), kw=dict(n_rows=n_rows, n_cols=n_cols))
     f_func = f"[eye({n_rows=}, {n_cols=})]"
-    for i in range(n_rows):
-        for j in range(_n_cols):
+    k = kw.get("k", 0)
+    expected = xp.asarray([[1 if j - i == k else 0
+                            for j in range(_n_cols)] for i in range(n_rows)]).reshape(n_rows, _n_cols)
+    assert out.shape == expected.shape
+    if xp.any(out != expected):
+        for i, j in zip(*xp.nonzero(out != expected)):
```

Comment:
As [`xp.nonzero`](https://data-apis.org/array-api/draft/API_specification/generated/array_api.nonzero.html?highlight=nonzero#array_api.nonzero) is optional on the spec, this whole branch of logic would need to check the [`--disable-data-dependent-shapes` flag](https://github.com/data-apis/array-api-tests/blob/37bbb580975b21a40c23463d651830dfc4dd35d0/conftest.py#L42-L48) we have before we use this for assertions... there's a way to do this in pytest but forgot how :sweat_smile: If dds is indeed disabled, we can default to the previous logic.